test/currencies/currencies.api-test.ts

```diff
@@ -0,0 +1,169 @@
+      expect(body).toEqual({
+        statusCode: 404,
+        message: 'The selected exchange rate is not available',
+        error: 'Not Found',
+      });
+    });
+
+    it('should return 400 on GET `/currencies/exchange?source=GBP&target=abc`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?source=GBP&target=abc');
+      expect(status).toBe(400);
+      expect(body).toEqual({
+        statusCode: 400,
+        message: ['target must be a valid ISO4217 currency code'],
+        error: 'Bad Request',
+      });
+    });
+
+    it('should return 400 on GET `/currencies/exchange?source=abc&target=AED`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?source=abc&target=AED');
+      expect(status).toBe(400);
+      expect(body).toEqual({
+        statusCode: 400,
+        message: ['source must be a valid ISO4217 currency code'],
+        error: 'Bad Request',
+      });
+    });
+
+    it('should return 400 on GET `/currencies/exchange?source=abc&target=def`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?source=abc&target=def');
+      expect(status).toBe(400);
+      expect(body).toEqual({
+        statusCode: 400,
+        message: ['source must be a valid ISO4217 currency code', 'target must be a valid ISO4217 currency code'],
+        error: 'Bad Request',
+      });
+    });
+
+    it('should return 400 on GET `/currencies/exchange?source=GBP`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?source=GBP');
+      expect(status).toBe(400);
+      expect(body).toEqual({
+        statusCode: 400,
+        message: ['target must be a valid ISO4217 currency code'],
+        error: 'Bad Request',
+      });
+    });
+
+    it('should return 400 on GET `/currencies/exchange?target=GBP`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?target=GBP');
+      expect(status).toBe(400);
+      expect(body).toEqual({
+        statusCode: 400,
+        message: ['source must be a valid ISO4217 currency code'],
+        error: 'Bad Request',
+      });
+    });
+  });
+
+  describe('/currencies', () => {
+    it('should return 200 on GET `/currencies`', async () => {
+      const { status, body } = await api.get('/currencies');
+      expect(status).toBe(200);
+      expect(body).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            id: expect.any(Number),
+            name: expect.any(String),
+            isoCode: expect.any(String),
+            created: expect.any(String),
+            updated: expect.any(String),
+            effectiveFrom: expect.any(String),
+            effectiveTo: expect.any(String),
+            acbsCode: expect.any(String),
+          }),
+        ]),
+      );
+    });
+  });
+
+  describe('/currencies/{isoCode}', () => {
+    it('should return 200 on GET `/currencies/GBP`', async () => {
```

Comment:
Test case where endpoint invoked is `/currencies/` please.

Reply:
that's already covered above?